src/pytorch_lightning/utilities/device_parser.py

```diff
@@ -330,3 +333,27 @@ def parse_hpus(devices: Optional[Union[int, str, List[int]]]) -> Optional[int]:
         raise MisconfigurationException("`devices` for `HPUAccelerator` must be int, string or None.")
 
     return int(devices) if isinstance(devices, str) else devices
+
+
+def num_cuda_devices() -> int:
+    """Returns the number of GPUs available.
+
+    Unlike :func:`torch.cuda.device_count`, this function will do its best not to create a CUDA context to for fork
+    support, if the platform allows it.
+    """
+    if not hasattr(os, "fork"):
+        return torch.cuda.device_count()
+    with multiprocessing.get_context("fork").Pool(1) as pool:
+        return pool.apply(torch.cuda.device_count)
+
+
+def is_cuda_available() -> bool:
+    """Returns a bool indicating if CUDA is currently available.
+
+    Unlike :func:`torch.cuda.is_available`, this function will do its best not to create a CUDA context to for fork
```

Comment:
If we didn't do this. Would there be failures or bugs or is it just a matter of leaving leftover CUDA context using memory?

Reply:
If we didn't do this, we would see an immediate error upon launching processes:

> RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method